### PR TITLE
Exclude callable field defaults from the OpenAPI schema

### PR DESCRIPTION
**Pull request by arttuperala**
## Description of the Change

Implements [the same sort of callable field.default exclusion that DRF implements](https://github.com/encode/django-rest-framework/blob/master/rest_framework/schemas/openapi.py#L538), as having a callable field default in the OpenAPI schema results in broken YAML output.

```yaml
field:
  type: string
  writeOnly: true
  default: !!python/object:rest_framework.fields.CreateOnlyDefault
    default: null
```

## Checklist

- [x] PR only contains one change (considered splitting up PR)
- [x] unit-test added
- [ ] documentation updated
- [x] `CHANGELOG.md` updated (only for user relevant changes)
- [x] author name in `AUTHORS`

### CODE REVIEW

File: tests/test_schema.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from rest_framework_json_api.schemas.openapi import AutoSchema
```
Comment (sliverc) [inline tests/test_schema.py:1]: Great that you added a test to our new tests folder.

What we try to do here is that the test file is in the same module structure as the python file it is testing.

So this file should be `tests/schemas/test_openapi.py`

File: tests/test_schema.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+from tests.serializers import CallableDefaultSerializer
+
+
+def test_schema_callable_default():
```
Comment (sliverc) [inline tests/test_schema.py:5]: If the test is testing part of a class we also use a class to group the test.

So the test would look like this:

```
class TestAutoSchema:
    def test_map_serializer_with_callable_default(self):
        ...
```

**Review by sliverc — APPROVED** [review]
Thanks for this addition. Well spotted. See my inline comments for some small adjustments.

File: tests/test_schema.py (hunk 1 above)
Comment (arttuperala) [reply tests/test_schema.py]: Moved.

File: tests/test_schema.py (hunk 2 above)
Comment (arttuperala) [reply tests/test_schema.py]: Punted into a class.

**Comment by sliverc** [comment]
Great. Thanks for your work. Merged.

**Pull request merged**